sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -231,10 +251,36 @@ export class AssessmentController {
         jobLogger.error(message);
         return job.error(ASSESSMENT_NOT_FOUND, message);
       }
-      await this.studentAssessmentService.updateAssessmentStatusAndSaveWorkflowData(
-        job.variables.assessmentId,
-        job.variables.workflowData,
-      );
+      // The updateAssessmentStatusAndSaveWorkflowData and assessImpactedApplicationReassessmentNeeded are executed in the same transaction
+      // to force then to be successfully executed together or to fail together. In this way the worker can be safely retried from Camunda.
+      await this.dataSource.transaction(async (entityManager) => {
+        // Update status and workflow data ensuring that it will be updated only for the first time to ensure the worker idempotency.
+        const updated =
```

Comment:
Can have a brief comment to mention that the updated also represents the execution of whole transaction(updateAssessmentStatusAndSaveWorkflowData and assessImpactedApplicationReassessmentNeeded) as per 254.

Reply:
@guru-aot @lewischen-aot @andrepestana-aot please let me know if the comments are clear or if please suggest something to improve.

Reply:
Yes it is clear, but can we rename the variable to sometihng called as updateFirstTime or something?

Reply:
`updatedFirstTime` will not represent what the variable means.
We can rename it to something like `alreadyUpdated` but it is also what the "update" in the past is trying to say.